src/ds/dll.rs

```diff
@@ -0,0 +1,196 @@
+/// A simple doubly-linked list for use in the `Lru`
```

Comment:
this is the main structure powering the `Lru` now, a simple doubly linked list, sharded 256x and protected by a `Mutex`, so contention isn't too high.